### PR TITLE
seperate js.js classes to their own files

### PR DESCRIPTION
**Pull request by th3fallen**
Seperates the tons of classes in js.js into their own files.

Also moves namespace var to a bootstrap file to be more global.
(oca may possibly need to be in there as well not sure though)

@PVince81 suggestions/comments

### CODE REVIEW

File: core/js/core.json [hunk 1]
```diff
@@ -15,10 +15,17 @@
 		"placeholder.js"
 	],
 	"modules": [
-		"compatibility.js",
-		"jquery.ocdialog.js",
-		"oc-dialogs.js",
+        "bootstrap.js",
+        "compatibility.js",
+        "jquery.ocdialog.js",
+        "oc-dialogs.js",
```
Comment (MorrisJobke) [inline core/js/core.json]: Intendation

File: core/js/history.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
+/**
+ * @author Clark Tomlinson  <clark@owncloud.com>
```
Comment (PVince81) [inline core/js/history.js]: I wrote this class, not you :wink:

**Comment by PVince81** [comment]
We should also split the unit tests. Currently the tests from js.js are in the file coreSpec.js: https://github.com/owncloud/core/blob/master/core/js/tests/specs/coreSpec.js

**Comment by PVince81** [comment]
Otherwise I agree that these classes needed to be separated :smile:

**Comment by th3fallen** [comment]
> We should also split the unit tests. Currently the tests from js.js are in the file coreSpec.js: https://github.com/owncloud/core/blob/master/core/js/tests/specs/coreSpec.js

agreed, i'll look inot that

**Comment by DeepDiver1975** [comment]
@PVince81 what shall we do with this?

**Comment by MorrisJobke** [comment]
> @PVince81 what shall we do with this?

Kill this and do it step by step and with a shorter round trip time (as in reviewing/merging faster) to not kill the work on the mentioned classes.

**Comment by PVince81** [comment]
I agree with @MorrisJobke. If you do it all at once there will always be conflicts.
Better do it one by one.

**Pull request closed without merging**